File: Unity Project/SuperEngineer/Assets/Scripts/PlayerControl.js
#pragma strict


private var moveSpeed : float = 7;
private var jumpSpeed:	float = 15;

var distToGround : float;

function Start () {
	// Getting the distance from the center to the ground.
	//distToGround = collider.bounds.extents.height /2 ;
distToGround = 2.5f;
}

function Update () {

rigidbody.velocity.x = moveSpeed;



if( (Input.GetKeyDown(KeyCode.W)) && (IsGrounded () || IsGrounded1 () || IsGrounded2 () )                ) {
animation.Play("Jump", PlayMode.StopAll);
rigidbody.velocity.y = jumpSpeed;

}
else if(IsGrounded ()|| IsGrounded1 () || IsGrounded2 ())
{
	animation.Play("RunCycle", PlayMode.StopSameLayer);	
}

}
function IsGrounded () : boolean { //Check if we are on the ground. Return true if we are else return null.

		return Physics.Raycast(transform.position, -Vector3.up, distToGround + 0.1);
	
}
function IsGrounded1 () : boolean { //Check if we are on the ground. Return true if we are else return null.

		return Physics.Raycast(transform.position + Vector3(-1,-.2,0), -Vector3.up, distToGround + 0.1);
}
function IsGrounded2 () : boolean { //Check if we are on the ground. Return true if we are else return null.

		return Physics.Raycast(transform.position + Vector3(1,-.2,0), -Vector3.up, distToGround + 0.1);	
}

File: Unity Project/SuperEngineer/Assets/Scripts/PlayerControl.cs
using UnityEngine;
using System.Collections;

public class PlayerControl : MonoBehaviour 
{
	private bool _isJumping = false;
	public float jumpForce;
	public float velocity;
	public static PlayerControl PlayerInstance;

	public bool IsGrounded 
	{
		get { return Physics.Raycast (transform.position, -Vector3.up, 2f); }
	}

	void Awake()
	{
		animation ["Jump"].speed = 0.5f;
		PlayerInstance = this;
		GapDetector.OnPlayerCollision += OnPlayerCollidedWithGap;
	}

	void OnDisable ()
	{
		GapDetector.OnPlayerCollision -= OnPlayerCollidedWithGap;
	}

	void Update ()
	{
		if (!_isJumping) 
		{
			animation.CrossFade ("RunCycle", 0.2f);
		}
		else 
		{
			animation.CrossFade ("Jump", 0.2f);
		}

		transform.position += Vector3.right * Time.deltaTime * velocity;

		if (Input.GetKeyDown (KeyCode.W) && this.IsGrounded) 
		{
			StartCoroutine (Jump ());
		}
	}

	IEnumerator Jump ()
	{
		if (!_isJumping) 
		{
			_isJumping = true;
			rigidbody.velocity = new Vector3 (0, jumpForce, 0);
			yield return new WaitForSeconds (1);
			while (!this.IsGrounded)
			{
				yield return new WaitForEndOfFrame ();
			}
			_isJumping = false;
		}
	}

	void OnPlayerCollidedWithGap ()
	{
		StartCoroutine (OnDie ());
	}

	IEnumerator OnDie ()
	{
		this.enabled = false;
		yield return new WaitForSeconds (1);
		GameOver.OnGameOver ();
	}
}
